proposals/OpaqueReflect.md

```diff
@@ -0,0 +1,147 @@
+# Specification of opaque reflection
+
+This spec addresses #1892. The goal is to permit the reflection of functions which contain symbols that are not already in the logic, through the automatic lifting of the symbols which are not in the logic, as uninterpreted functions.
+
+The idea behind issue #1892 is that you may want to reflect a function without reflecting all the functions that are in the definition. For example, `Data.Map.lookup` may not be necessarily reflected for a function containing it to be reflected and reasoned about. Currently, the process of adding a dummy measure for functions we don't want to reflect is done manually and is often required. Such that it would be preferable to make opaque-reflection a default behavior, if no reflection is available for the symbol. In particular, because it allows the reflection of functions using imported symbols, that may not have been reflected there in the first place.
```

Comment:
Maybe precede this first mention of measures with a sentence at the end of the first paragraph:

> In this document we use "uninterpreted functions" and "measures" interchangeably.